src/fsharp/vs/ServiceUntypedParse.fs

```diff
@@ -351,12 +351,24 @@ type FSharpParseFileResults(errors : FSharpErrorInfo[], input : Ast.ParsedInput
             (fun () -> 
                 let locations = findBreakPoints()
                 
-                match locations |> List.filter (fun m -> rangeContainsPos m pos) with
-                | [] ->
-                    match locations |> List.filter (fun m -> rangeBeforePos m pos |> not) with
-                    | [] -> Seq.tryHead locations
-                    | locationsAfterPos -> Seq.tryHead locationsAfterPos
-                | coveringLocations -> Seq.tryLast coveringLocations)
+                if pos.Column = 0 then
+                    // we have a breakpoint that was set with mouse at line start
+                    match locations |> List.filter (fun m -> m.StartLine = m.EndLine && pos.Line = m.StartLine) with
+                    | [] ->
+                        match locations |> List.filter (fun m -> rangeContainsPos m pos) with
```

Comment:
I assume we change that soon. I don't want to make it DRY yet. Can you please still merge it?